File: leetcode/100/P100.rb
# 100. Same Tree
# Definition for a binary tree node.
# class TreeNode
#     attr_accessor :val, :left, :right
#     def initialize(val)
#         @val = val
#         @left, @right = nil, nil
#     end
# end

# @param {TreeNode} p
# @param {TreeNode} q
# @return {Boolean}
def is_same_tree(p, q)
	if !p && !q
		true
	elsif p && q
		p.val == q.val && is_same_tree(p.left, q.left) && is_same_tree(p.right, q.right)
	else
		false
	end
end

File: leetcode/100/src/P100.java
/**
 * Created by qwordy on 2/22/16.
 * 100. Same Tree
 */

class TreeNode {
	int val;
	TreeNode left;
	TreeNode right;
	TreeNode(int x) { val = x; }
}

class Solution {
	public boolean isSameTree(TreeNode p, TreeNode q) {
		return p == null && q == null || p != null && q != null && p.val == q.val && isSameTree(p.left, q.left) && isSameTree(p.right, q.right);
	}
}

public class P100 {
}
